src/storage.rs

```diff
@@ -36,16 +70,94 @@ impl Storage {
     /// # Description
     /// Closes all file descriptors
     /// # Examples
+
+    // @TODO specify more useful error type
     pub fn close(&mut self) -> Result<(), ()> {
         // @TODO implement
         Ok(())
     }
+
+    /// # Description
+    /// Blobs count contains closed blobs and one active, if is some
+    /// # Examples
+    /// ```
+    /// use pearl::Builder;
+    ///
+    /// let mut storage = Builder::new().work_dir("/tmp/pearl/").build();
+    /// storage.init();
+    /// assert_eq!(storage.blobs_count(), 1);
+    /// ```
+    pub fn blobs_count(&self) -> usize {
+        self.blobs.len() + if self.active_blob.is_some() { 1 } else { 0 }
+    }
+}
+
+impl Storage {
+    fn prepare_work_dir(&self) -> io::Result<()> {
+        let path = Path::new(&self.config.work_dir);
+        if !path.exists() {
+            debug!("creating work dir recursively: {}", path.display());
+            fs::create_dir_all(path)
+        } else {
+            debug!("work dir exists: {}", path.display());
+            Ok(())
+        }
+    }
+
+    // @TODO specify more useful error type
+    #[inline]
+    fn init_active_blob(&mut self) -> Result<(), ()> {
+        self.active_blob = Box::new(Some(Default::default()));
+        Ok(())
+    }
+
+    // @TODO specify more useful error type
+    fn init_new(&mut self) -> Result<(), ()> {
+        self.init_active_blob()
+    }
+
+    // @TODO specify more useful error type
+    fn init_from_existing(&mut self, files: Vec<DirEntry>) -> Result<(), ()> {
+        self.blobs = files
+            .iter()
+            .filter_map(|entry| {
+                // @TODO implement more file validations
+                if entry.metadata().ok()?.is_file() {
+                    Some(Blob::from_file(entry.path()).ok()?)
+                } else {
+                    debug!("skipping file \"{}\"", entry.path().display());
+                    None
+                }
+            })
+            .collect();
+        if let Some(blob_index) = self.search_for_active_blob() {
+            self.active_blob = Box::new(Some(self.blobs.remove(blob_index)));
+            Ok(())
+        } else {
+            self.init_active_blob()
+        }
+    }
+
+    fn search_for_active_blob(&self) -> Option<usize> {
+        // @TODO Search for last active blob by index extracted from file name
+        let mut active_blob_id = None;
+        self.blobs.iter().enumerate().find(|(i, blob)| {
+            let res = blob.is_active();
```

Comment:
Do the blob need to know if he is active or not? I suggest to determine the active blob as the last one in the directory.

Reply:
Fair point about who knows about the state of a blob, 'cause `Storage` contains configuration and can check the state.
Second, if the last blob is full, a new one must be initialized